data_model/src/lib.rs

```diff
@@ -663,11 +719,27 @@ where
 }
 
 /// This trait marks entity that implement it as identifiable with an `Id` type to find them by.
-pub trait Identifiable: Debug + Clone {
+pub trait Identifiable: Debug {
     /// Type of entity's identification.
-    type Id: Into<IdBox> + Debug + Clone + Eq + Ord;
-    /// Type used to register `Identifiable` entity
-    type RegisteredWith: Into<RegistrableBox>;
+    type Id: Into<IdBox> + fmt::Display + fmt::Debug + Clone + Eq + Ord;
+
+    /// Return reference to the type's identification.
+    fn id(&self) -> &Self::Id;
+}
+
+/// Trait that marks the entity as having metadata.
+pub trait HasMetadata {
+    // type Metadata = metadata::Metadata;
+    // Uncomment when stable.
+
+    /// The metadata associated to this object.
+    fn metadata(&self) -> &metadata::Metadata;
+}
+
+/// Trait for objects that are registered by proxy.
+pub trait Registered: Identifiable {
+    /// The type that is used to register this entity with.
```

Comment:
```suggestion
    /// The type that is used to register this entity.
```

don't really need the `with` here